packages/dashboard-frontend/src/widgets/organic-sessions/compare.js

```diff
@@ -59,10 +59,15 @@ export const useOrganicSessionsCompare = ( dataProvider, remoteDataProvider, dat
 	 * @returns {Promise<OrganicSessionsCompareData|Error>} The promise of OrganicSessionsCompareData or an Error.
 	 */
 	const getOrganicSessionsCompare = useCallback( ( options ) => {
+		const widgetName = "organicSessionsCompare";
 		return remoteDataProvider.fetchJson(
 			dataProvider.getEndpoint( "timeBasedSeoMetrics" ),
-			{ options: { widget: "organicSessionsCompare" } },
-			options );
+			{
+				options: { widget: widgetName },
+			},
+			options,
+			widgetName
```

Comment:
Repeating a comment from the other side. This makes it so the widget needs to know it needs the cached provider variant. It would be nice if it was not aware what remote provider it received.

+ comment applies to all